geo/src/algorithm/intersects/rect.rs

```diff
@@ -40,7 +40,33 @@ where
 symmetric_intersects_impl!(Rect<T>, Point<T>);
 symmetric_intersects_impl!(Rect<T>, MultiPoint<T>);
 
-symmetric_intersects_impl!(Rect<T>, Polygon<T>);
+impl<T> Intersects<Polygon<T>> for Rect<T>
+where
+    T: GeoNum,
+{
+    fn intersects(&self, rhs: &Polygon<T>) -> bool {
+        /*
+        sufficient to show that any of these are true:
+        some corner of the polygon intersects the rectangle
+        some corner of the rectangle intersects the polygon
+        some edge of polygon (interior or exterior) intersects edge of rectangle
+        */
+
+        if has_disjoint_bboxes(self, rhs) {
+            return false;
+        }
+
+        // rhs.intersects(pt) is the most expensive, so check it last
+        // only required if rectangle sits entirely within polygon
+        rhs.coords_iter().any(|p| self.intersects(&p))
+            || rhs.lines_iter().any(|l| {
+                self.lines_iter()
+                    .any(|other_line| l.intersects(&other_line))
+            })
+            || self.min().intersects(rhs)
```

Comment:
What's this `self.min` check about? 

I guess we've shown above that none of the lines intersect, and that the polygon isn't within the rect, so if there __is__ an intersection, the rect must be fully within the polygon. In which case, there's no need to check *all* the rect coords, if any one of them intersects, that's sufficient to show that the rect is entirely within the polygon. Is that right?

I got there eventually, but it's pretty dense. You've commented the cases nicely at the top of the function, but I suspect it'd be easier to follow if you re-organized the code down into a case by case analysis with the comments in line.